community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/RecordFormats.java

```diff
@@ -44,9 +48,15 @@ public Factory( String key, String... altKeys )
 
     String storeVersion();
 
+    /**
+     * @return format generation, with the intent of usage being that a store can migrate to a newer or
+     * same generation, but not to an older generation.
+     */
+    int generation();
```

Comment:
it's to be able to tell whether or not we're trying to do an upgrade or a downgrade... given two formats. Is there a better way?